docs/devsite-help/rest-libraries.md

```diff
@@ -0,0 +1,168 @@
+# HTTP/1.1 + REST libraries
+
+As described in the [library types page](library-types.md), the
+HTTP/1.1 + REST libraries are API-specific libraries with source code in
+the [google-api-dotnet-client repository](https://github.com/googleapis/google-api-dotnet-client),
+generated from [Discovery documents](https://developers.google.com/discovery/v1/getting_started)
+instead of API protobuf representations. API calls are made over HTTP/1.1 with
+JSON content.
+
+These libraries depend heavily on the support libraries of Google.Apis and
+Google.Apis.Core, which have their source code in the same repository.
+
+These libraries are still supported and automatically updated as the API
+surfaces change, but the underlying code and generator is not being actively worked
+on, and only high-impact feature requests are considered.
+
+Some features are not as discoverable as we'd like them to be; this page
+describes the use of these features.
+
+## Service object reuse
+
+Service client classes (e.g. `StorageService`) are thread-safe and reusable.
+Typically your application should create a single instance and use it for all operations.
+
+By default, every instantiation of a service client class creates a new instance of
+[HttpClient](https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpclient),
+and creating many of these is, in general, a bad idea. See, for example,
+[this post](https://aspnetmonsters.com/2016/08/2016-08-27-httpclientwrong/) for why.
+
+Every instantiation of a client class also uses more CPU time and requires more memory allocation,
+which are generally best avoided if possible.
+
+Sometimes it *is* required to create an instance per request; for example, if different
```

Comment:
Actually, Apiary does cleanly support per-call credentials.

So I wonder if it's best to say, create a single client, and if you need to pass different credentials per call, do this...

Reply:
I haven't included the "and if you need to pass different credentials per call" part as I wasn't sure exactly what you meant, but I've just removed the example. (It does beg the question about the circumstances for when that's appropriate though.)

Reply:
(Couldn't include this as part of the review).

The most common case for needing a different client is when you need different credentials. But in the case of Apiary, service clients support per call credentials cleanly. I think we can add a section on "Per call credentials" (now, or later) and here simply not specify other corner cases for which you may need different clients.